tools/reference_checker.cpp

```diff
@@ -210,10 +210,13 @@ int run(const ReferenceCheckerOptions &options) {
                                        << node->getSourceInfo().toPosition().sourceLine << ": "
                                        << sourceFragment;
             } else {
-                // FIXME: any elegant solution to just print first line of `replaced->toString()`?
-                flayOptimizationOutput << "Replaced node at line "
-                                       << node->getSourceInfo().toPosition().sourceLine << ": "
-                                       << sourceFragment << " with " << replaced->toString();
+                BUG_CHECK(replaced->getSourceInfo().isValid(),
```

Comment:
Do not use `BUG_CHECK`, instead use `ASSIGN_OR_RETURN`

Reply:
Sorry I didn't quite get how to use`ASSIGN_OR_RETURN` here. In my understanding, `ASSIGN_OR_RETURN` checks if a pointer-like variable, and returns if it is null, otherwise, dereferences it. But we don't have a pointer-like variable here.

What about just doing something similar to [this](https://github.com/fruffy/flay/blob/c001584e041bb03fa0648b17f3a3d3c43c9a28b8/tools/reference_checker.cpp#L222)?

Reply:
Oh I meant "RETURN_IF_FALSE", apologize for the confusion.